### ISSUE TITLE
Correct way to set custom audio. Audio not playing

### ISSUE THREAD

**Issue by ahrherrera** [phonegap/phonegap-plugin-push#2562]
### Expected Behaviour
Custom notification sound should play when notification arrives 
### Actual Behaviour
It does not play the custom sound when notification arrives 
### Reproduce Scenario (including but not limited to)
Sending a custom audio file name from GCM it does not play the custom audio
#### Steps to Reproduce 
Add the resource file
<resource-file src="assets/sound/sound.mp3" target="app/src/main/res/raw/sound.mp3" />
*But at running this line disappears from config.xml

and Send the push notification from GCM with the value sound:sound.mp3
#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)
Android 8
#### (Android) What device vendor (e.g. Samsung, HTC, Sony...)
Samsung
#### Cordova CLI version and cordova platform version

    cordova --version                                    # e.g. 6.0.0
8.0.0
    cordova platform version android                     # e.g. 4.1.1
Installed platforms:
  android 7.0.0
Available platforms:
  browser ~5.0.1
  ios ~4.5.4
  osx ~4.0.1
  windows ~5.0.0
  www ^3.12.0

#### Plugin version

    cordova plugin version | grep phonegap-plugin-push   # e.g. 1.5.3
phonegap-plugin-push 2.1.3 "PushPlugin"

#### Sample Push Data Payload
this is the data received:

 ```
{"sound":"sound.mp3",
"title":"Tienes una nueva orden cerca de Ti",
"message":"Nueva Orden",
"additionalData":{"idOrden":"1","coldstart":false,"foreground":true}}
```

#### Sample Code that illustrates the problem
PushOptions

```
const options: PushOptions = {
      android: {
        senderID: '658380658535',
        sound: true
      },
      ios: {
        alert: 'true',
        badge: true,
        sound: false
      },
      windows: {}
    };
```

**Comment by macdonst** [comment]
@ahrherrera you shouldn't send the file extension for the sound file. Run `adb logcat | grep Push` to see if there are any issues when the notification comes in.

**Comment by ahrherrera** [comment]
Hi again.

I changed the payload data, but still does not sound the audio file, the resource-file tag:
<resource-file src="src/assets/sounds/colibri.mp3" target="app/src/main/res/raw/colibri.mp3" />

`{"sound":"colibri",
"title":"Tienes una nueva orden cerca de Ti",
"message":"Nueva Orden",
"additionalData":{
"idPedido":"1",
"fecha-pedido":"27/09/2018 4:40PM",
"lat":"12.139913",
"lng":"-86.275305",
"coldstart":false,
"foreground":true
}
}`

**Comment by macdonst** [comment]
@ahrherrera yeah but without a log from the device I can't tell what the plugin is picking up.

**Comment by emc2spider** [comment]
@ahrherrera I created a similar ticket a while ago [(https://github.com/phonegap/phonegap-plugin-push/issues/2017)](https://github.com/phonegap/phonegap-plugin-push/issues/2017) so far the only way I've found to make the notifications play custom sounds is to register them to a custom channel and add the sound file reference there, if you have more than one custom sound you'll need as many custom channels as well I guess. That's what I'm doing anyway, is not ideal but still haven't found a more elegant solution to this problem.

**Comment by cikcoh** [comment]
Hi, how to set custom sound in ios, mine not working, it just play the default sound,
i have tried with both payload
`     "registration_ids" : ["...JwvLRxWQ:APA...."],
     "notification":{
     	"title": "postman",
     	 "body": "postman",
     	 "sound": "police_siren"
     },
     "aps" : {
     "alert" : "something",
     "sound" : "police_siren.caf"
    },
     "data": {
        "notId": 12, 
      }`
or just
`     "registration_ids" : ["..vLRxWQ:APA91bG...."],
     "notification":{
     	"title": "postman",
     	 "body": "postman",
     	 "sound": "police_siren"
     },
     "data": {
        "notId": 12, 
      }`
i have the file in platform/ios/myapp/Resources/

**Comment by kaiesh** [comment]
> @ahrherrera I created a similar ticket a while ago [(https://github.com/phonegap/phonegap-plugin-push/issues/2017)](https://github.com/phonegap/phonegap-plugin-push/issues/2017) so far the only way I've found to make the notifications play custom sounds is to register them to a custom channel and add the sound file reference there, if you have more than one custom sound you'll need as many custom channels as well I guess. That's what I'm doing anyway, is not ideal but still haven't found a more elegant solution to this problem.

I'm using the FCM Legacy HTTP API to target individual devices, but have got this working to play custom sounds on each notification. I have summarised my APK design here:
https://github.com/phonegap/phonegap-plugin-push/issues/2492#issuecomment-463498852

Hope it helps you!

**Comment by kaiesh** [comment]
> Hi, how to set custom sound in ios, mine not working, it just play the default sound,
> i have tried with both payload
> ` "registration_ids" : ["...JwvLRxWQ:APA...."], "notification":{ "title": "postman", "body": "postman", "sound": "police_siren" }, "aps" : { "alert" : "something", "sound" : "police_siren.caf" }, "data": { "notId": 12, }`
> or just
> ` "registration_ids" : ["..vLRxWQ:APA91bG...."], "notification":{ "title": "postman", "body": "postman", "sound": "police_siren" }, "data": { "notId": 12, }`
> i have the file in platform/ios/myapp/Resources/

After I completed the implementation of my Android sound implementation detailed at https://github.com/phonegap/phonegap-plugin-push/issues/2492#issuecomment-463498852 I then began work on iOS custom sounds.

It turns out the process only requires the copying of sounds to the appropriate resource location. I did this with this config.xml entry (note I use the PhoneGap Build cloud service):
```
<platform name="ios">
   ...
   <resource-file src="resources/common/sounds/mysound.mp3" target="Sounds/mysound.mp3" />
   ...
</platform>
```
_Please refer to the linked comment above on my ZIP file structure in regards to the sounds directory._

Then to trigger the custom sound via FCM (using HTTP Legacy API), I simply specify `mysound.mp3`. The guide advises not to include the file extension, but I find that this is the only way to play the sound on iOS. Conveniently - the Android playback is not affected if you supply the file extension.

I am building using plugin version `2.0.0`, and with PGB `cli-7.0.1` for iOS devices, and `cli-7.1.0` for Android devices.

**Comment by Vallo** [comment]
I'm facing the same problem. My alert.mp3 is stored on /res/raw/alert.mp3. 

This is the output of adb logcat:

```
09-17 17:52:06.206 22325 22532 D Push_FCMService: onMessage - from: 258671441928
09-17 17:52:06.208 22325 22532 D Push_FCMService: sender id = 258671441928
09-17 17:52:06.208 22325 22532 D Push_FCMService: normalize extras
09-17 17:52:06.208 22325 22532 D Push_FCMService: key = body
09-17 17:52:06.208 22325 22532 D Push_FCMService: replace key body with message
09-17 17:52:06.209 22325 22532 D Push_FCMService: no locale found for key = message, error Value myAppName of type java.lang.String cannot be converted to JSONObject
09-17 17:52:06.209 22325 22532 D Push_FCMService: key = sound
09-17 17:52:06.209 22325 22532 D Push_FCMService: replace key sound with sound
09-17 17:52:06.209 22325 22532 D Push_FCMService: key = title
09-17 17:52:06.209 22325 22532 D Push_FCMService: replace key title with title
09-17 17:52:06.209 22325 22532 D Push_FCMService: no locale found for key = title, error Value Notifications of type java.lang.String cannot be converted to JSONObject
09-17 17:52:06.210 22325 22532 D Push_FCMService: background
09-17 17:52:06.210 22325 22532 D Push_FCMService: message =[myAppName push notifications working successfully!]
09-17 17:52:06.210 22325 22532 D Push_FCMService: title =[Notifications test]
09-17 17:52:06.210 22325 22532 D Push_FCMService: contentAvailable =[null]
09-17 17:52:06.210 22325 22532 D Push_FCMService: forceStart =[null]
09-17 17:52:06.210 22325 22532 D Push_FCMService: create notification
09-17 17:52:06.210 22325 22532 E Push_FCMService: Number format exception - Error parsing notId: s == null
09-17 17:52:06.219 22325 22532 D Push_FCMService: Using channel ID = PushPluginChannel
09-17 17:52:06.224 22325 22532 D Push_FCMService: stored icon=pushicon
09-17 17:52:06.224 22325 22532 D Push_FCMService: stored iconColor=red
09-17 17:52:06.224 22325 22532 D Push_FCMService: stored sound=true
09-17 17:52:06.224 22325 22532 D Push_FCMService: stored vibrate=true
09-17 17:52:06.224 22325 22532 D Push_FCMService: using icon from plugin options
09-17 17:52:06.225 22325 22532 D Push_FCMService: android.resource://com.myAppName.app/raw/alert.mp3
09-17 17:52:06.231 22325 22532 D Push_FCMService: create actions: with in-line
```

default sound is being played.
